msal/application.py

```diff
@@ -198,8 +198,9 @@ def __init__(
                 authority or "https://login.microsoftonline.com/common/",
                 self.http_client, validate_authority=validate_authority)
             # Here the self.authority is not the same type as authority in input
+        self.client = None
         self.token_cache = token_cache or TokenCache()
-        self.client = self._build_client(client_credential, self.authority)
+        self.client_credential = client_credential
```

Comment:
```suggestion
        self._client_credential = client_credential
```
We won't change pre-existing property names, but from now on, let's be more conscious on using leading underscore on internal property names.